crates/host/src/wasmbus/providers/http_server/address.rs

```diff
@@ -1,37 +1,32 @@
 use core::net::SocketAddr;
-use core::str::FromStr as _;
 
 use std::collections::HashMap;
 use std::sync::Arc;
 
-use anyhow::{bail, Context as _};
+use anyhow::Context as _;
 use http::header::HOST;
 use http::uri::Scheme;
 use http::Uri;
 use http_body_util::BodyExt as _;
-use hyper_util::rt::{TokioExecutor, TokioIo};
-use nkeys::XKey;
-use tokio::sync::{broadcast, Mutex, RwLock};
-use tokio::task::JoinSet;
+use tokio::sync::{Mutex, RwLock};
+use tokio::task::JoinHandle;
 use tokio::time::Instant;
-use tracing::{error, info_span, instrument, trace_span, Instrument as _, Span};
-use wasmcloud_core::InterfaceLinkDefinition;
-use wasmcloud_provider_http_server::{default_listen_address, load_settings, ServiceSettings};
-use wasmcloud_provider_sdk::provider::{
-    handle_provider_commands, receive_link_for_provider, ProviderCommandReceivers,
-};
-use wasmcloud_provider_sdk::{LinkConfig, LinkDeleteInfo, ProviderConnection};
+use tracing::{info_span, instrument, trace_span, Instrument as _, Span};
+use wasmcloud_provider_http_server::{load_settings, ServiceSettings};
+use wasmcloud_provider_sdk::{LinkConfig, LinkDeleteInfo};
 use wasmcloud_tracing::KeyValue;
 use wrpc_interface_http::ServeIncomingHandlerWasmtime as _;
 
 use crate::wasmbus::{Component, InvocationContext};
 
-struct Provider {
-    address: SocketAddr,
-    components: Arc<RwLock<HashMap<String, Arc<Component>>>>,
-    links: Mutex<HashMap<Arc<str>, HashMap<Box<str>, JoinSet<()>>>>,
-    lattice_id: Arc<str>,
-    host_id: Arc<str>,
+use super::listen;
+
+pub(crate) struct Provider {
+    pub(crate) address: SocketAddr,
+    pub(crate) components: Arc<RwLock<HashMap<String, Arc<Component>>>>,
+    pub(crate) links: Mutex<HashMap<Arc<str>, HashMap<Box<str>, JoinHandle<()>>>>,
```

Comment:
I'd just keep the value as `JoinSet<()>` for simplicity. Otherwise, we need to implement `Drop` on `Provider` to clear the map

Reply:
Good call, after re-reading the `Drop` implementation for `JoinSet<()>` I think it makes a lot more sense to use it.